sdk/storage/Azure.Storage.Common/src/Shared/StorageConnectionString.cs

```diff
@@ -1037,78 +920,51 @@ private static (Uri, Uri) ConstructQueueEndpoint(string scheme, string accountNa
         /// <param name="accountName">The name of the storage account.</param>
+        /// <param name="hostNamePrefix">Prefix that appears before the host name, e.g. "blob".</param>
         /// <param name="endpointSuffix">The Endpoint DNS suffix; use <c>null</c> for default.</param>
-        /// <returns>The default table endpoint.</returns>
-        private static (Uri, Uri) ConstructTableEndpoint(string scheme, string accountName, string endpointSuffix)
+        /// <param name="sasToken">The sas token; use <c>null</c> for default.</param>
+        /// <returns></returns>
+        private static (Uri, Uri) ConstructUris(
+            string scheme,
+            string accountName,
+            string hostNamePrefix,
+            string endpointSuffix,
+            string sasToken)
         {
-            if (string.IsNullOrEmpty(scheme))
-            {
-                throw Errors.ArgumentNull(nameof(scheme));
-            }
-
-            if (string.IsNullOrEmpty(accountName))
-            {
-                throw Errors.ArgumentNull(nameof(accountName));
-            }
+            var primaryUriBuilder = new UriBuilder
+            {
+                Scheme = scheme,
+                Host = string.Format(
+                        CultureInfo.InvariantCulture,
+                        "{0}.{1}.{2}",
+                        accountName,
+                        hostNamePrefix,
+                        endpointSuffix),
+                Query = sasToken
+            };
 
-            if (string.IsNullOrEmpty(endpointSuffix))
-            {
-                endpointSuffix = Constants.ConnectionStrings.DefaultEndpointSuffix;
-            }
+            var secondaryUriBuilder = new UriBuilder
+            {
+                Scheme = scheme,
+                Host = string.Format(
+                        CultureInfo.InvariantCulture,
+                        "{0}{1}.{2}.{3}",
+                        accountName,
+                        Constants.ConnectionStrings.SecondaryLocationAccountSuffix,
+                        hostNamePrefix,
+                        endpointSuffix),
+                Query = sasToken
+            };
 
-            var primaryUri = string.Format(
-                CultureInfo.InvariantCulture,
-                "{0}://{1}.{2}.{3}/",
-                scheme,
-                accountName,
-                Constants.ConnectionStrings.DefaultTableHostnamePrefix,
-                endpointSuffix);
-
-            var secondaryUri = string.Format(
-                CultureInfo.InvariantCulture,
-                "{0}://{1}{2}.{3}.{4}",
-                scheme,
-                accountName,
-                Constants.ConnectionStrings.SecondaryLocationAccountSuffix,
-                Constants.ConnectionStrings.DefaultTableHostnamePrefix,
-                endpointSuffix);
-
-            return (new Uri(primaryUri), new Uri(secondaryUri));
+            return (primaryUriBuilder.Uri, secondaryUriBuilder.Uri);
```

Comment:
Nit - is it worth sharing a single builder and just changing the host between Uri creation?